lib/utils/src/graph/algorithms.cc

```diff
@@ -251,6 +273,28 @@ std::unordered_set<Node> get_sources(DiGraphView const &g) {
   return sources;
 }
 
+std::unordered_set<Node> get_sinks(DiGraphView const &g) {
+  std::unordered_set<Node> dsts;
+  for (Node const &n : get_nodes(g)) {
+    auto outgoing = get_outgoing_edges(g, n);
+    if (outgoing.size() == 0) {
+      dsts.insert(n);
+    }
+  }
+  return dsts;
+}
+
+std::unordered_set<Node> get_sinks(MultiDiGraphView const &g) {
+  std::unordered_set<Node> dsts;
```

Comment:
Maybe change to use `filter` from `containers.h`? Would shorten this quite a bit I think